plugins/inputs/sqlserver/azuresqlqueries.go

```diff
@@ -382,6 +382,116 @@ HAVING
 OPTION(RECOMPILE);
 `
 
+const sqlAzureDBPartQueryPeriod = `
+DECLARE @QueryData NVARCHAR(MAX) = ?1; --input parameter
+
+DECLARE @lastIntervalEndTimestamp BIGINT = 0;
+
+IF @QueryData != '' AND @QueryData IS NOT NULL
+	SET @lastIntervalEndTimestamp = CAST(@QueryData AS BIGINT);
+
+DECLARE @magicNumberDays BIGINT = 4294967296, @magicNumberSeconds BIGINT = 300, @secondsInDay INT = 86400;
+
+DECLARE @timeGrain BIGINT = (SELECT interval_length_minutes * 60 FROM sys.database_query_store_options WITH (NOLOCK)); --use query store time grain
+
+DECLARE @currDate DATETIME = GETUTCDATE();
+DECLARE @currIntervalEndTimestamp BIGINT = CONVERT(BIGINT, DATEDIFF(day, 0, @currDate)) * @secondsInDay + DATEDIFF(second, DATEADD(day, DATEDIFF(day, 0, @currDate), 0), @currDate) / @timeGrain * @timeGrain;
```

Comment:
Can you add comments of what you are trying to achieve here?  I know has to do with interval config but right now reviewing the code isn't easy

Reply:
Added comments